pkg/dynamicconfig/reflectors.go

```diff
@@ -4,54 +4,118 @@
 package dynamicconfig
 
 import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"slices"
+	"strings"
+
 	"github.com/cilium/statedb"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
+	apivalidation "k8s.io/apimachinery/pkg/util/validation"
 
+	"github.com/cilium/cilium/pkg/annotation"
 	"github.com/cilium/cilium/pkg/k8s"
+	ciliumv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
 	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
 	"github.com/cilium/cilium/pkg/k8s/utils"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option/resolver"
 )
 
-var (
-	CiliumConfigMap          = "cilium-config"
-	CiliumConfigMapNamespace = "kube-system"
-
-	// Lower number means higher priority
-	// When adding a new source, the source priority needs to be updated here
-	priorities = map[string]int{
-		CiliumConfigMap: 0,
-	}
+const (
+	metadataName = "metadata.name="
 )
 
-func NewConfigMapReflector(cs k8sClient.Clientset, t statedb.RWTable[DynamicConfig], c config) []k8s.ReflectorConfig[DynamicConfig] {
+func NewConfigMapReflector(cs k8sClient.Clientset, t statedb.RWTable[DynamicConfig], c config, l *slog.Logger) []k8s.ReflectorConfig[DynamicConfig] {
 	if !cs.IsEnabled() || !c.EnableDynamicConfig {
 		return []k8s.ReflectorConfig[DynamicConfig]{}
 	}
+	sources, overrides, err := parseConfigs(c)
+	if err != nil {
+		l.Error("Processing configs", logfields.Error, err)
+		return []k8s.ReflectorConfig[DynamicConfig]{}
+	}
+
+	processedSources := map[resolver.ConfigSource]struct{}{} // Ensure no source duplicates
```

Comment:
Not sure why this is needed? Wouldn't it make better sense to dedup in parseConfigs and return a set?

Reply:
Deduped the resolver. thanks